code/modules/reagents/reagent_containers/hypospray_vr.dm

```diff
@@ -22,3 +22,44 @@
 	..()
 	for (var/i = 1 to 7)
 		new /obj/item/weapon/reagent_containers/hypospray/autoinjector/miner(src)
+
+/obj/item/weapon/reagent_containers/hypospray/science
+	name = "prototype hypospray"
+	desc = "This reproduction hypospray is nearly a perfect replica of the early model DeForest hyposprays, sharing many of the same features. However, there are additional safety measures installed to prevent unwanted injections."
+
+/obj/item/weapon/reagent_containers/hypospray/science/attack(mob/living/M as mob, mob/user as mob)
+	if(!reagents.total_volume)
+		to_chat(user, "<span class='warning'>[src] is empty.</span>")
+		return
+	if (!istype(M))
+		return
+
+	var/mob/living/carbon/human/H = M
+	if(istype(H))
+		var/obj/item/organ/external/affected = H.get_organ(user.zone_sel.selecting)
+		if(!affected)
+			to_chat(user, "<span class='danger'>\The [H] is missing that limb!</span>")
+			return
+		else if(affected.robotic >= ORGAN_ROBOT)
+			to_chat(user, "<span class='danger'>You cannot inject a robotic limb.</span>")
+			return
+
+		if(H != user)
+			to_chat(user, "<span class='notice'>You begin injecting [H] with \the [src].</span>")
+			to_chat(H, "<span class='danger'> [user] is trying to inject you with \the [src]!</span>")
+			if(!do_after(user, 30))
```

Comment:
This might have been copied from the original, but is bad.

The person can walk away and you will still inject them. You must pass the target as the 3rd parameter.